torchtune/models/llama4/_chunked_attention.py

```diff
@@ -0,0 +1,57 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+from torchtune.modules.attention_utils import (
+    _MaskType,
+    _SUPPORTS_FLEX_ATTENTION,
+    causal_mask_flex,
+)
+
+if _SUPPORTS_FLEX_ATTENTION:
+    from torch.nn.attention.flex_attention import BlockMask, create_block_mask
+
+
+def get_chunked_attention_mask(
+    mask: Optional[_MaskType],
+    chunk_size: int,
+    bsz: int,
+    seq_len: int,
+) -> _MaskType:
+    """ """
+    # TODO: check this somewhere that doesn't get called every forward
+    if not _SUPPORTS_FLEX_ATTENTION:
+        raise ValueError("Local attention is only supported with flex attention.")
+    if mask is None:
+        mask_mod = causal_mask_flex
+        q_seq_len, kv_seq_len = seq_len, seq_len
+    elif isinstance(mask, BlockMask):
+        mask_mod = mask.mask_mod
+        q_seq_len, kv_seq_len = mask.seq_lengths
+    else:
+        raise ValueError("Unsupported mask type")
+
+    def chunked_attention_mask_mod(
```

Comment:
glad this works for you :)

Reply:
@drisspg yes, thank you! The paste you shared was super helpful